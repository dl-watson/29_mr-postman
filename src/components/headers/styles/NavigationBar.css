.NavigationBar {
    display: flex;
    background-color: cornflowerblue;
    border: 1px dotted red;

    a {
        text-decoration: none;
        color: black;
    }

    .navButton {
        margin: 5px;
        padding: 10px;
        border: none;
        cursor: pointer;
        background-color: white;
    }

    .navButton::hover{
        background-color: lightcoral;
    }
}
