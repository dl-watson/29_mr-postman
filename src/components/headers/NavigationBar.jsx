import React from "react";
import { BrowserRouter as Router, Link } from "react-router-dom";
import styles from "./styles/NavigationBar.css";

const NavigationBar = () => {
  return (
    <Router>
      <div className={styles.NavigationBar}>
        <Link to="/">
          <div className={styles.navButton}>home</div>
        </Link>
        <Link to="/history">
          <div className={styles.navButton}>history</div>
        </Link>
        <Link to="/help">
          <div className={styles.navButton}>help</div>
        </Link>
      </div>
    </Router>
  );
};

export default NavigationBar;
