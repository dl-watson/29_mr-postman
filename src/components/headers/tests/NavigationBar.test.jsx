import React from "react";
import { render, cleanup } from "@testing-library/react";
import NavigationBar from "../NavigationBar";

describe("NavigationBar component", () => {
  afterEach(() => cleanup());
  it("renders NavigationBar", () => {
    const { asFragment } = render(<NavigationBar />);
    expect(asFragment()).toMatchSnapshot();
  });
});
