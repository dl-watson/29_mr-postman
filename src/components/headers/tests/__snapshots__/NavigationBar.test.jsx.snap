// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NavigationBar component renders NavigationBar 1`] = `
<DocumentFragment>
  <div
    class="NavigationBar"
  >
    <a
      href="/"
    >
      <div
        class="navButton"
      >
        home
      </div>
    </a>
    <a
      href="/history"
    >
      <div
        class="navButton"
      >
        history
      </div>
    </a>
    <a
      href="/help"
    >
      <div
        class="navButton"
      >
        help
      </div>
    </a>
  </div>
</DocumentFragment>
`;
